lib/internal/util/comparisons.js

```diff
@@ -502,7 +502,7 @@ function setEquiv(a, b, strict, memo) {
     for (const val of b) {
       // Primitive values have already been handled above.
       if (typeof val === 'object' && val !== null) {
-        if (!setHasEqualElement(set, val, strict, memo))
+        if (!a.has(val) && !setHasEqualElement(set, val, strict, memo))
           return false;
       } else if (!strict &&
```

Comment:
Will this else-if condition ever be reached? If `!a.has(val) && !setHasEqualElement(set, val, strict, memo)` is false then certainly `!strict && !a.has(val) && !setHasEqualElement(set, val, strict, memo)` would be false?

Reply:
Which `if` are you matching with which `else`?

Reply:
The else will be reached as it's the inner if. It could be optimized code wise (there is some duplication that can be prevented), but that could also be done in a later PR to check what implementation is best.

Reply:
@lemire  Oops, my bad, I was matching the "else" with the wrong "if", the one on line 505. Indentation is hard 😜